### PR TITLE
CIV-12577 Notification for HwF more infor need

### PR DESCRIPTION
**Pull request by ElifDenizEsea**
**Before creating a pull request make sure that:**

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-12577

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotificationData.java [hunk 1]
```diff
@@ -87,6 +87,9 @@ public interface NotificationData {
 
     // evidence upload
     String UPLOADED_DOCUMENTS = "uploaded documents";
+    String HwF_MORE_INFO_DATE = "date";
+    String HwF_MORE_INFO_FEE_TYPE = "typeOfFee";
```
[the remark this answers is not in the document]
Comment (ElifDenizEsea) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotificationData.java:91]: done

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyLiPClaimantHwFOutcomeHandler.java [hunk 2]
```diff
@@ -136,6 +142,28 @@ CLAIMANT_NAME, getPartyNameBasedOnType(caseData.getApplicant1()),
         );
     }
 
+    private Map<String, String> getMoreInformationProperties(CaseData caseData) {
+        HelpWithFeesMoreInformation moreInformation = new HelpWithFeesMoreInformation();
```
[the remark this answers is not in the document]
Comment (ElifDenizEsea) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyLiPClaimantHwFOutcomeHandler.java:146]: done

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotificationData.java [hunk 3]
```diff
@@ -87,6 +87,9 @@ public interface NotificationData {
 
     // evidence upload
     String UPLOADED_DOCUMENTS = "uploaded documents";
+    String HwF_MORE_INFO_DATE = "date";
```
[the remark this answers is not in the document]
Comment (ElifDenizEsea) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotificationData.java:90]: done

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/MoreInformationHwfCallbackHandler.java [hunk 4]
```diff
@@ -67,7 +73,22 @@ private CallbackResponse validationMoreInformation(CallbackParams callbackParams
...
+        }
+        if (caseData.isHWFTypeClaimIssued()) {
+            HelpWithFeesDetails claimIssuedHwfDetails = caseData.getClaimIssuedHwfDetails();
+            updatedData.claimIssuedHwfDetails(claimIssuedHwfDetails.toBuilder().hwfCaseEvent(MORE_INFORMATION_HWF).build());
+        }
+
         return AboutToStartOrSubmitCallbackResponse.builder()
+            .data(updatedData.build().toMap(objectMapper))
             .build();
+
```
[the remark this answers is not in the document]
Comment (ElifDenizEsea) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/MoreInformationHwfCallbackHandler.java:92]: done

File: src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/CcdDashboardClaimantClaimMatcher.java [hunk 5]
```diff
@@ -311,6 +311,13 @@ public boolean isPaymentPlanRejected() {
             && caseData.hasApplicantRejectedRepaymentPlan());
     }
 
+    @Override
+    public boolean isHwFMoreInformationNeeded() {
+        return ((null != caseData.getHelpWithFeesMoreInformationClaimIssue()
```
[the remark this answers is not in the document]
Comment (ElifDenizEsea) [reply src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/CcdDashboardClaimantClaimMatcher.java:316]: done

File: src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/DashboardClaimStatus.java [hunk 6]
```diff
@@ -6,6 +6,9 @@
 
 public enum DashboardClaimStatus {
 
+    HWF_MORE_INFORMATION_NEEDED(
+        Claim::isHwFMoreInformationNeeded
```
[the remark this answers is not in the document]
Comment (ElifDenizEsea) [reply src/main/java/uk/gov/hmcts/reform/civil/model/citizenui/DashboardClaimStatus.java:10]: done

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/MoreInformationHwfCallbackHandlerTest.java [hunk 7]
```diff
@@ -69,11 +79,34 @@ void shouldValidationMoreInformationHearing_withInvalidDate() {
...
+            CaseData caseData = CaseData.builder()
+                .claimIssuedHwfDetails(hwfeeDetails)
+                .hwfFeeType(
+                    FeeType.CLAIMISSUED)
+                .build();
+            //When
+            CallbackParams params = callbackParamsOf(caseData, ABOUT_TO_SUBMIT);
+            var response = (AboutToStartOrSubmitCallbackResponse) handler.handle(params);
+            //Then
+            assertThat(response).isNotNull();
```
[the remark this answers is not in the document]
Comment (ElifDenizEsea) [reply src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/MoreInformationHwfCallbackHandlerTest.java:97]: done

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyLiPClaimantHwFOutcomeHandlerTest.java [hunk 8]
```diff
@@ -335,4 +429,20 @@ private Map<String, String> getNotificationDataMapUpdateRefNumberHearing() {
             );
         }
     }
+
+    private List<HwFMoreInfoRequiredDocuments> getMoreInformationDocumentList() {
+        List<HwFMoreInfoRequiredDocuments> docList = new ArrayList<>();
+        docList.add(HwFMoreInfoRequiredDocuments.CHILD_MAINTENANCE);
```
[the remark this answers is not in the document]
Comment (ElifDenizEsea) [reply src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/notification/NotifyLiPClaimantHwFOutcomeHandlerTest.java:435]: done

**Review by manish14836 — APPROVED** [review]

**Review by neeta-hmcts — APPROVED** [review]

**Pull request merged**